### DISCUSSION TITLE
🙏 Help:  Angular Reactive Forms

### DISCUSSION THREAD

**Discussion by marcelhdl** [public-ui/kolibri#7010] [Q&A] [answered]
### Summary

Hallo liebe Community,

ist es derzeit möglich das folgende Beispiel einer einfachen Angular Reactive Form mit Kolibri Mitteln umzusetzen?
Wir haben hier schon einige Versuche hinter uns und kommen derzeit nicht weiter.
Auf den ersten Blick werden die aus Angular Sicht notwendigen Properties nicht an das input-Element weitergereicht.
Wir haben versucht das ganze mittels kol-form und kol-input-text umzusetzen.
```    
<kol-form className="w-full" [formGroup]="form" [_on]="eventHandler">
      <kol-input-text _id="input1" _name="input1" formControlName="firstName2" _label="Input 1" />
      <kol-input-text _id="input2" _name="input2" formControlName="lastName2" _label="Input 2" />
      <kol-button _type="submit" _label="Submit" />
</kol-form>
```
FUp:  https://angular.dev/guide/forms/reactive-forms

### Example

https://stackblitz.com/edit/stackblitz-starters-jnh2df?file=src%2Fmain.ts

**Comment by deleonio** [comment]
Hallo @marcelhdl.

vielen Dank für die Anfrage. Ich würde @sdvg bitten, die Frage zu beantworten. Er hat vor Kurzem an einer Optimierung der Framework- und Lib-Integration gearbeitet.

Viele Grüße
Martin

**Comment by sdvg** [comment, marked as answer]
Das Thema wurde gestern in der CoP besprochen. Hier noch einmal die Stichpunkte:
- Im ersten Schritt entwickeln wir native DOM-Events für alle Komponenten, die analog zu den Callbacks im `_on`-Objekt genutzt werden können. Das sollte den Komfort bei der Framework-Nutzung schon einmal erhöhen.
- Gleichzeitig bereiten wir damit eine bessere Nutzung der Framework-Adapter (Angular und Vue) vor. Hier sollten wir noch einen Versuch starten, die [ValueAccessors korrekt zu definieren](https://github.com/ionic-team/stencil-ds-output-targets/blob/main/packages/angular/README.md), und damit `ngModel` nutzbar zu machen. Idealerweise würde es dann auch direkt mit Reactive Forms funktionieren, aber das wäre noch zu prüfen.

Referenzen:
https://github.com/public-ui/kolibri/issues/6987 - aktuelles Ticket für die DOM-Events
https://github.com/public-ui/kolibri/pull/6953 - Pilot-PR, in dem auch bereits die Adapter-Konfiguration angetestet wurde.
